client/client.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+)
+
+// HTTPClient provides a properly configured http.Client object used to send
+// requests out to a ContainerPilot process's control socket.
+type HTTPClient struct {
+	http.Client
+	socketPath string
+}
+
+var socketType = "unix"
+
+func socketDialer(socketPath string) func(string, string) (net.Conn, error) {
+	return func(_, _ string) (net.Conn, error) {
+		return net.Dial(socketType, socketPath)
+	}
+}
+
+// NewHTTPClient initializes an client.HTTPClient object by configuring it's
+// socketPath for HTTP communication through the local file system.
+func NewHTTPClient(socketPath string) (*HTTPClient, error) {
+	if socketPath == "" {
+		err := errors.New("control server not loading due to missing config")
+		return nil, err
+	}
+
+	client := &HTTPClient{}
+	client.Transport = &http.Transport{
+		Dial: socketDialer(socketPath),
+	}
+
+	return client, nil
+}
+
+// Reload makes a request to the reload endpoint of a ContainerPilot process.
+func (c HTTPClient) Reload() (*http.Response, error) {
+	resp, err := c.Post("http://control/v3/reload", "application/json", nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		mesg := fmt.Sprintf("expected 200 but got %v\n%+v", resp.StatusCode, resp)
```

Comment:
There's probably an argument that could be made to return only an `error` object, but until there are consumers it's all handwavy. So returning the status code sounds like the right move now.